### DISCUSSION TITLE
docs/invoke

### DISCUSSION THREAD

**Discussion by giscus** [STranslate/STranslate#77] [Announcements]
# docs/invoke

WPF 开发的一款即开即用、即用即走的翻译(OCR)工具

https://stranslate.zggsong.com/docs/invoke.html

**Comment by liangshaojie** [comment]
如何使用呢

**Reply by ZGGSONG** [reply]
有示例，用http请求

```
curl http://127.0.0.1:50020/translate -d "你好世界"

curl http://127.0.0.1:50020/translate

#其他的path查看上面的介绍
```

**Comment by GD2021** [comment]
能否写一个quicker指令，快速调用

**Reply by ZGGSONG** [reply]
不好意思，我没用过quicker，暂时可能没时间研究这块了。如果你有兴趣可以写一写结合quicker的指令，大家一起构建完善开源项目也是更好的发展方向，[文档仓库](https://github.com/ZGGSONG/STranslate-doc)，如果有想了解软件更深入的部分可以通过以下方式联系我：
1. [mail](mailto:zggsong@foxmail.com)
2. [qq group](https://github.com/ZGGSONG/STranslate?tab=readme-ov-file#discussion)

**Reply by HYL-Wah** [reply]
https://getquicker.net/Sharedaction?code=7ab58ae9-c486-4fc5-6486-08dd652533b7

**Comment by eliot-ye** [comment]
启动服务失败请重新配置端口

打开外部服务调用就报上面这个怎么搞？换端口也不行

**Reply by ZGGSONG** [reply]
请确认更换端口是否被占用 [端口占用查询](https://blog.csdn.net/algebra007/article/details/119242167)

```shell
netstat -ano | findstr 50020
```

**Reply by eliot-ye** [reply]
已确认是端口问题，我换了4开头的才可以

**Comment by zhangyiming748** [comment]
有没有更详细一点的接口文档？这个太简单了看不明白啊，就比如get的时候带的参数名是什么啊？

**Reply by ZGGSONG** [reply]
没写参数的就是不要参数，写了就是要

**Reply by zhangyiming748** [reply]
但是我得提供要翻译的内容啊
比如Get http://127.0.0.1:50020/translate&参数=要翻译的内容啊

**Reply by ZGGSONG** [reply]
> 没写参数的就是不要参数

没写就是不支持参数，要填翻译内容用 post 加参数

<img width="994" height="196" alt="图片" src="https://github.com/user-attachments/assets/cd1789f5-d07d-46d2-81de-97f115dde2a2" />

**Reply by zhangyiming748** [reply]
我还是没太明白，restful 的 get 或者 post 方法 按照我浅陋的理解应该有个`参数名`去接收`你好世界`啊
比如
```golang
package main

import (
    "fmt"
    "net/http"
)

func handler(w http.ResponseWriter, r *http.Request) {
    // 解析查询参数
    err := r.ParseForm()
    if err != nil {
        http.Error(w, "无法解析查询参数", http.StatusBadRequest)
        return
    }
    
    // 获取参数值
    name := r.URL.Query().Get("name") // 假设查询参数是name
    age := r.URL.Query().Get("age")    // 假设查询参数是age

    // 响应结果
    fmt.Fprintf(w, "姓名: %s, 年龄: %s", name, age)
}

func main() {
    // 注册路由
    http.HandleFunc("/", handler)

    // 启动HTTP服务器
    fmt.Println("服务器在 :8080 启动")
    http.ListenAndServe(":8080", nil)
}
```
或者其他语言，得有个接受我需要翻译的内容的变量吧
我写postman请求的时候他也得要我给一个参数名啊
我不是做前端的 不懂还有没有其他前端的高级玩法

**Reply by ZGGSONG** [reply]
简单来说  我get都不接受参数 只有post接收参数

**Reply by ZGGSONG** [reply]
```CSharp
//Please use GET like `curl localhost:50020/translate -d \"hello world\"`"
switch (request.HttpMethod)
{
    case "GET":
        ExecuteExternalCall(ecAction, "");
        break;

    case "POST":
        ExecuteExternalCall(ecAction, content);
        break;

    default:
        throw new Exception("Method Not Allowed");
}
```

**Comment by vae5201314** [comment]
你好,为什么我截图翻译后,然后把图片里面内容也识别出来可以吗,我还想复制源文本,现在怎么没有这个功能了,之前是另外开一个窗口的

**Reply by ZGGSONG** [reply]
设置里把图片内翻译关掉就行，软件最顶上右侧按钮里左边那个是个快捷入口

**Comment by vae5201314** [comment]
我输入的API KEY,但是还是调用不起来呢,为什么

**Reply by ZGGSONG** [reply]
配置上以后开始直接翻译输出框里有错误信息

**Comment by SUXHK** [comment]
你好开发者，当设置成失焦不关闭窗口时，能否添加个关闭按钮呢

**Reply by ZGGSONG** [reply]
<img width="998" height="758" alt="图片" src="https://github.com/user-attachments/assets/dc4d14cc-2997-4c71-9c9f-203ccdadc12c" />

**Reply by dexzhou** [reply]
失去焦点这个对于某些人是很好用的，但是很多时候，很多用户，都习惯了右上角的x，建议加个x

**Reply by ZGGSONG** [reply]
<img width="1750" height="1330" alt="图片" src="https://github.com/user-attachments/assets/d385178d-9f2c-47d5-8ce9-ee34d743fbea" />

**Comment by JAY13205130** [comment]
能否做一个与「有道翻译」的截图翻译功能，它是框选界面后，直接在界面位置做翻译，这样可以更加直观

**Reply by ZGGSONG** [reply]
https://stranslate.zggsong.com/docs/#%E5%9B%BE%E7%89%87%E7%BF%BB%E8%AF%91-%E7%B2%BE%E7%AE%80%E7%AA%97%E5%8F%A3

**Comment by gongjunjie42-cell** [comment]
作者 怎么设置 驼峰首字母前面添加空格  有道驼峰命名 没有空格翻译不了 
例如：
AccountQuotaExceeded 这样翻译出的就还是英文
Account Quota Exceeded 这样即可正常翻译

**Reply by ZGGSONG** [reply]
> AccountQuotaExceeded 这样翻译出的就还是英文

大概率是某次错误的缓存，Ctrl+Enter即可刷新缓存

**Reply by gongjunjie42-cell** [reply]
没有用  不加空格 翻译不出 我用的是有道

**Reply by dexzhou** [reply]
https://getquicker.net/Sharedaction?code=77125e7e-8483-4f68-19b3-08df07def119 曲线救国，我用qcuiker搞一下

**Reply by ZGGSONG** [reply]
<img width="681" height="897" alt="图片" src="https://github.com/user-attachments/assets/13d0b3c6-dbac-467d-b57b-e3ae964765fa" />
<img width="681" height="897" alt="图片" src="https://github.com/user-attachments/assets/6bd2350a-fc6b-406c-8838-3e40010aa67e" />
<img width="1235" height="948" alt="图片" src="https://github.com/user-attachments/assets/f4eab6c9-72ff-41b5-86b7-302041c94774" />

**Reply by dexzhou** [reply]
每个人的需求是不一样的，大驼峰复制出来的没有空格，对于程序员可能很好用，但是对于做外贸的就不友好了，比如发品到阿里国际站平台的时候，要求每个单词首字母大写，你搞个没有空格的，没法用。
既然你把这个软件都做成插件的了，想必是秉着开放的态度，后期有可能还有第三方插件，扩展之类的。

**Reply by ZGGSONG** [reply]
> 你搞个没有空格的，没法用。

> 既然你把这个软件都做成插件的了，想必是秉着开放的态度，后期有可能还有第三方插件，扩展之类的。

1. 首先，这个功能跟插件系统没关系，
2. 其次，开源免费软件，满足自用是第一要务，没有使用场景，不能预判所有人的预判，
3. 最后，我何时说过大家提的功能不做的？有需求提 feature issue。一般来说，小众需求超过一定数量的人提我都尽量满足了。

**Reply by ZGGSONG** [reply]
软件功能有你喜欢的，你用的开心就好，没有的，我也没有义务必须开发
我最烦颐指气使的，提需求的要明白，你不为我的时间支付任何酬劳！

**Reply by dexzhou** [reply]
哟哟哟，生气了，我还真的为你支付了，微信还给你留言了，提了大驼峰的事儿。自己看看吧。前几天的事儿。你是开发者你说了算，我们也是提提建议。以为是开放的心态，你高兴就好。

**Reply by dexzhou** [reply]
再说一下，我不止为您的工具支付过，其它的软件，有打赏功能的，我一般也都会支持一下，期待您的新版本。

**Reply by ZGGSONG** [reply]
哟哟哟   我求您给我支付的？   支付了我就要给你开发  您老谁呀   爱用不用  出门右拐  滚一边去 影响老子心情
